codegen/lib/graphql_java_gen.rb

```diff
@@ -280,11 +281,25 @@ def java_doc(element)
     unless element.description.nil?
       description = wrap_text(element.description, 100)
       description = description.chomp("\n").gsub("\n", "\n* ")
-      doc << "/**\n"
       doc << '* '
       doc << description
-      doc << "\n*/"
     end
+
+    if element.respond_to?(:deprecated?) && element.deprecated?
+      if (doc.length > 0)
+        doc << "\n*"
+        doc << "\n*"
+      else
+        doc << '*'        
+      end
+      doc << ' @deprecated '
```

Comment:
but if you are asking if this annotation applicable for enum values the answer yes.